portal-impl/src/com/liferay/portal/events/ServicePreAction.java

```diff
@@ -907,12 +907,8 @@ private ThemeDisplay _initThemeDisplay(
 			realUser = UserLocalServiceUtil.getUserById(realUserId.longValue());
 		}
 
-		if (!user.isActive()) {
-			user = company.getGuestUser();
-
-			if (realUser == null) {
-				httpSession.invalidate();
-			}
+		if ((realUser == null) || !realUser.isActive()) {
```

Comment:
The original condition should remain the same (`!user.isActive()`) and the operator should be `&&`

Reply:
@MirnaGama, I believe why we need to keep the `!user.isActive()` check and the `user = company.getGuestUser()` check is the case of impersonation. When impersonation is going on the user object is the impersonated user and if the impersonated has been deactivated in the meantime we need to check that and take the necessary step.